docs/concepts/what-is-an-activity-heartbeat.md

```diff
@@ -28,3 +28,26 @@ That way if a Worker fails it can be handled in a timely manner.
 
 A Heartbeat can include an application layer payload that can be used to _save_ Activity Execution progress.
 If an [Activity Task Execution](/concepts/what-is-an-activity-task-execution) times out due to a missed Heartbeat, the next Activity Task can access and continue with that payload.
+
+**What Activities should Heartbeat?**
+
+Heartbeating is best thought about not in terms of time, but in terms of "How do you know you are making progress"?
+For short-term operations, progress updates are not a requirement. However, checking the progress and status of Activities that run over long periods is almost always useful.
+
+Consider the following when deciding on setting Activity Hearbeats:
+
+- Your underlying task must be able to report definite progress.
+  Note that your Workflow cannot read this progress information while the Activity is still executing (or it would have to store it in Event History).
+  You may report progress to external sources if you need it exposed to the user.
```

Comment:
```suggestion
Consider the following when setting Activity Hearbeats:

- Your underlying task must be able to report definite progress.
  Note that your Workflow cannot read this progress information while the Activity is still executing (or it would have to store it in Event History).
  You can report progress to external sources if you need it exposed to the user.
```